clients/src/main/java/org/apache/kafka/clients/NetworkClient.java

```diff
@@ -51,7 +54,13 @@
     private final Selectable selector;
     
     private final MetadataUpdater metadataUpdater;
-
+    
+    /* a list of nodes we've connected to in the past */
+    private final List<Integer> nodesEverSeen;
+    private final Map<Integer, Node> nodesEverSeenById;
```

Comment:
Right, with the `LinkedHashMap` we add two additional references per `Map.Entry` instance when compared to a `HashMap`.